src/platform_impl/macos/view.rs

```diff
@@ -686,6 +768,23 @@ extern "C" fn key_down(this: &Object, _sel: Sel, event: id) {
         };
 
         if pass_along {
+            // Use IME
+            let is_ime_activated: bool = *this.get_ivar("isIMEActivated");
+            if is_ime_activated && is_arrow_or_space_key(virtual_keycode.unwrap()) {
+                (*this).set_ivar("isPreediting", true);
+                let marked_text_ref: &mut id = this.get_mut_ivar("markedText");
+                let composed_string = marked_text_ref.clone().string();
+                let slice = slice::from_raw_parts(
+                    composed_string.UTF8String() as *const c_uchar,
+                    composed_string.len(),
+                );
+                let composed_string = str::from_utf8_unchecked(slice);
+                delete_marked_text(state, composed_string.chars().count() as i32);
```

Comment:
What is the reasoning behind deleting the IME text here? As far as I understand, we always delete the text in `set_marked_text` before sending the new one.

Reply:
That's true.
I misunderstood some points while debugging....